### PR TITLE
Fix parsing of storage-aggregation.conf patterns with special characters

### LINKED ISSUE #1990: Conf parsing changes fail to parse regex
**Describe the bug**

In #1989 some changes were made to the conf parse code. In particular [this commit](https://github.com/grafana/configparser/commit/f5c9e475c7f0f1f8ea64b45bd9c1d226d01d391f) seems to disallow any strings with `[]` in it. In our case, storage-aggregations has some settings like:

`pattern = ^[^;]+\.max(?:;|$)`

which used to parse fine and now we get:

`2021-06-25 09:18:24.738 [FATAL] can't read storage-aggregation file "/etc/metrictank/storage-aggregation.conf": invalid line "pattern = ^[^;]+\\.max(?:;|$)": [ and ] are reserved for section headers and this contains additional non-section header data`

If I comment out the lines from that commit, everything works as expected. Perhaps this check should only look at chars up to the first `=` (or the whole line if no `=` exists)?

**Helpful Information**
Metrictank Version: master
Golang Version: 1.15.11

### PR DESCRIPTION
**Pull request by Dieterbe**
This uses the patched config library (not merged yet) https://github.com/grafana/configparser/pull/1 (this PR includes more details, readme updates, test case updates, etc)

to fix #1990 

@shanson7 how does this look to you?

### CODE REVIEW

File: conf/aggregations_test.go [hunk 1]
```diff
@@ -91,31 +91,48 @@ func TestReadAggregations(t *testing.T) {
...
 						AggregationMethod: []Method{Max},
 					},
 				},
 				DefaultAggregation: defaultAggregation(),
 			},
 		},
+		{
+			title: "pattern with special characters",
+			in: `[foo]
+			pattern = [^ab](x|y)\p{Greek}.*,:;!@%&=+-_/?<>"'~$#ab
```
Comment (shanson7) [inline conf/aggregations_test.go:121]: This looks a good test for the issue I had.

File: vendor/github.com/grafana/configparser/configparser.go [hunk 2]
```diff
@@ -10,8 +10,8 @@
 //  - Options without values (ex: can be used to group a set of hostnames)
 //  - Options without a named section (ex: a simple option=value file)
 //  - Find sections with regexp pattern matching on section names, ex: dc1.east.webservers where regex is '.webservers'
-//  - # or ; as comment delimiter
-//  - = or : as value delimiter
+//  - # as comment delimiter
+//  - = as value delimiter
```
Comment (shanson7) [inline vendor/github.com/grafana/configparser/configparser.go:14]: Is this a behavior change?

**Review by shanson7 — COMMENTED** [review]
LGTM for my uses. Seems like support for `;` as a comment and `:` as a value delimiter was dropped, but I don't use those. Not sure about the fallout of that.

**Comment by Dieterbe** [comment]
>  Seems like support for ; as a comment and : as a value delimiter was dropped, but I don't use those. Not sure about the fallout of that.

yes. the config parser PR is a bit more explicit about it, and has some more unit tests, but basically for storage-schemas.conf and storage-aggregation.conf (our use case), those 2 chars never made sense. it's always been `=` and `#` and the only thing `:` and `:` were doing was introducing more undesired and unexpected behavior. i want to support people having `:` and `;` in their patterns if they choose to.

File: conf/aggregations_test.go (hunk 1 above)
Comment (Dieterbe) [reply conf/aggregations_test.go:121]: the configparser PR actually has a new unit test with your exact pattern, btw :)

**Review by shanson7 — APPROVED** [review]
LGTM

**Pull request merged**